src/shared/bg_public.h

```diff
@@ -29,6 +29,10 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 // bg_public.h -- definitions shared by both the server game and client game modules
 //==================================================================
 
+#include <glm/vec3.hpp>
+
+#define VEC2GLM( x ) glm::vec3( (x)[0], (x)[1], (x)[2] )
```

Comment:
Can this be glm::make_vec3( x );?

Reply:
It seems like it could. The pro of doing it that way is that it is also compatible with Vec3. I'm not too fond of having to rewrite every commit using Vec3.

This VEC2GLM macro is removed in the last commit of the branch anyway

Reply:
:shrug: Not something I feel strongly about, so I don't mind either way.

Reply:
Rewriting the branch would be annoying but maybe we can at least make it a function so the argument is not evaluated 3 times. `glm::vec3 VEC2GLM(const float* x) { ... }`

Reply:
I guess I could make two inline functions. On the other hand the compiler should already identify those (unless they are function calls, but I don't think so)

Reply:
Do you really need to rewrite the branch or can you just rewrite this single commit to update the function? All other commits should then reuse it. Vec3 is going to be deleted anyways.

Reply:
So I tried a bit, and after implementing it for Vec3, vec3_t and glm::vec3 (because it's sometimes called on itself, I get some `error: unknown type name 'Vec3'` somewhere. In the end I think it's just easier to keep the macro.

Reply:
Make it a template then. `template<typename T> glm::vec3 VEC2GLM(const T& x) { return {x[0],x[1],x[2]}; }`

Reply:
fair enough

Reply:
rebased